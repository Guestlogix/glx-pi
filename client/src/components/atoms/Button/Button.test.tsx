import * as React from "react"
import * as Enzyme from "enzyme"
import * as Adapter from "enzyme-adapter-react-16"
import { Button } from "."

Enzyme.configure({ adapter: new Adapter() })

describe("Button Basic functionality", () => {
  it("should render", () => {
    const wrapper = Enzyme.shallow(<Button />)
    expect(wrapper).toMatchSnapshot()
  })
})
