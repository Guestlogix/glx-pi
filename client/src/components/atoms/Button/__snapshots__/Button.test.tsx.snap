// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Basic functionality should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  &.ripple {
    background-position: center;
    transition: background 0.8s;
  }

  &.ripple:focus {
    outline: none;
  }

  &.ripple:active {
    background-color: ",
            [Function],
            ";
    background-size: 100%;
    transition: background 0s;
  }

  background: ",
            [Function],
            ";
  opacity: 1;
  border-radius: 7px;
  padding: 0.5rem 3rem;
  font-weight: bold;
  font-size: 17px;
  margin: 0 5rem;
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
  &.ripple {
    background-position: center;
    transition: background 0.8s;
  }

  &.ripple:focus {
    outline: none;
  }

  &.ripple:active {
    background-color: ",
              [Function],
              ";
    background-size: 100%;
    transition: background 0s;
  }

  background: ",
              [Function],
              ";
  opacity: 1;
  border-radius: 7px;
  padding: 0.5rem 3rem;
  font-weight: bold;
  font-size: 17px;
  margin: 0 5rem;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
